### PR TITLE
[Merged by Bors] - Adapt discovery to best practice

### PR DESCRIPTION
**Pull request by maltesander**
## Description

fixes https://github.com/stackabletech/trino-operator/issues/119

- Discovery now works via setting env variable from config map
- Improved the Trino MVP without HTTPS, Authentication, OPA etc.

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Review by razvan — COMMENTED** [review]
I wouldn't install os packages when setting up a Trino cluster because that can go wrong in many ways.

File: rust/operator-binary/src/controller.rs [hunk 1]
```diff
@@ -729,55 +648,136 @@ pub fn error_policy(_error: &Error, _ctx: Context<Ctx>) -> ReconcilerAction {
...
+        ),
+        _ => None,
+    })
 }
 
-async fn hive_connect(trino: &TrinoCluster, client: &Client) -> Result<Option<String>> {
-    let mut hive_connect_string = None;
+fn container_prepare_args() -> Vec<String> {
+    vec![[
+        "microdnf install openssl",
```
[the remark this answers is not in the document]
Comment (maltesander) [reply rust/operator-binary/src/controller.rs:675]: yeah agree will change.

**Comment by lfrancke** [comment]
> I wouldn't install os packages when setting up a Trino cluster because that can go wrong in many ways.

Yes! We should _never_ assume that those machines have internet access!

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**